lib/ui/views/login_view.dart

```diff
@@ -86,16 +142,32 @@ class _EmailTextField extends ProviderWidget<LoginViewModel> {
 }
 
 class _PasswordTextField extends ProviderWidget<LoginViewModel> {
+  final GlobalKey<FormState> formKey;
+  final TextEditingController controller;
+  final FocusNode currentFocusNode;
+  final FocusNode nextFocusNode;
+
+  _PasswordTextField({
+    this.formKey,
+    @required this.controller,
+    this.currentFocusNode,
+    this.nextFocusNode,
+  });
+
   @override
   Widget build(BuildContext context, LoginViewModel model) {
     final local = AppLocalizations.of(context);
 
     return TextFormField(
-      controller: model.passwordController,
+      controller: controller,
       validator: model.validatePassword,
-      focusNode: model.passwordFocusNode,
+      focusNode: currentFocusNode,
       obscureText: true,
       textInputAction: TextInputAction.send,
+      onChanged: (password) {
+        model.setPassword(password);
+        formKey.currentState?.validate();
```

Comment:
Either add the first ? back or make the formKey required

Reply:
formKey removed